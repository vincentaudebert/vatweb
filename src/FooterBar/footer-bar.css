.footer-bar {
  color: #fff;
  background-color: #007acc;
  font-size: 0.625rem; }

.footer-bar__info {
  margin-left: 0.625rem; }
  .footer-bar__info:first-child {
    margin-left: 0; }

.footer-bar--left {
  margin-left: 0.75rem !important; }

.footer-bar--right {
  float: right !important;
  margin-right: 0.75rem !important; }
