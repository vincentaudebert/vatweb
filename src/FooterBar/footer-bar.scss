@import '../css/variables';
@import '../css/mixins';

.footer-bar {
  color: $color-white;
  background-color: $color-blue;
  font-size: rem(10px);
}

.footer-bar__info {
  margin-left: rem(10px);
  &:first-child {
    margin-left: 0;
  }
}

.footer-bar--left {
  margin-left: $spacing !important;
}

.footer-bar--right {
  float: right !important;
  margin-right: $spacing !important;
}
