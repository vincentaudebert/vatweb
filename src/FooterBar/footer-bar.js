import React from 'react';
import './footer-bar.css';
import GoInfo from 'react-icons/lib/go/info';
import GoGitBranch from 'react-icons/lib/go/git-branch';
import GoCloudUpload from 'react-icons/lib/go/cloud-upload';
import GoCheck from 'react-icons/lib/go/check';
import GoAlert from 'react-icons/lib/go/alert';
import FaSmile from 'react-icons/lib/fa/smile-o';
import FaBell from 'react-icons/lib/fa/bell';
import TiDelete from 'react-icons/lib/ti/delete';

const FooterBar = props => {
  return (
    <div className="footer-bar row">
      <div className="col-4-sm left footer-bar--left">
        <a
          href="https://github.com/vincentaudebert/vatweb"
          target="_blank"
          className="reset-link footer-bar__info"
          title="Download my CV"
          rel="noopener noreferrer"
        >
          <GoGitBranch /> master (joke)
        </a>
        <span className="footer-bar__info">
          <GoCloudUpload />
        </span>
        <span className="footer-bar__info" title="Active Linters">
          <GoInfo /> ESLint, Standard
        </span>
        <span className="footer-bar__info" title="0 Warning">
          <TiDelete /> 0 <GoAlert /> 0
        </span>
      </div>
      <div className="col-4-sm" />
      <div className="col-4-sm right footer-bar--right">
        <span className="footer-bar__info">Spaces: 2</span>
        <span className="footer-bar__info">UTF-8</span>
        <span className="footer-bar__info">LF</span>
        <span className="footer-bar__info">JavaScript</span>
        <span className="footer-bar__info">
          Prettier:<GoCheck />
        </span>
        <a
          href="/assets/Vincent_Audebert_CV.pdf"
          target="_blank"
          className="reset-link footer-bar__info"
          title="Download my CV"
          rel="noopener noreferrer"
        >
          <FaSmile />
        </a>
        <span className="footer-bar__info" role="img" aria-label="Bell">
          <FaBell />
        </span>
      </div>
    </div>
  );
};

export default FooterBar;
