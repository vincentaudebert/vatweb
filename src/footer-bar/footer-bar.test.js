import React from 'react';
import renderer from 'react-test-renderer';
import FooterBar from './footer-bar';

describe('FooterBar', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<FooterBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
