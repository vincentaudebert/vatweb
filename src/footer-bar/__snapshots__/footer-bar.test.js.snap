// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterBar renders correctly 1`] = `
<div
  className="footer-bar row"
>
  <div
    className="d-inline-block left footer-bar__left"
  >
    <a
      className="reset-link footer-bar__info"
      href="https://github.com/vincentaudebert/vatweb"
      rel="noopener noreferrer"
      target="_blank"
      title="Download my CV"
    >
      <svg
        className="icon--footer"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 10 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 5c0-1.11-.89-2-2-2a1.993 1.993 0 0 0-1 3.72v.3c-.02.52-.23.98-.63 1.38-.4.4-.86.61-1.38.63-.83.02-1.48.16-2 .45V4.72a1.993 1.993 0 0 0-1-3.72C.88 1 0 1.89 0 3a2 2 0 0 0 1 1.72v6.56c-.59.35-1 .99-1 1.72 0 1.11.89 2 2 2 1.11 0 2-.89 2-2 0-.53-.2-1-.53-1.36.09-.06.48-.41.59-.47.25-.11.56-.17.94-.17 1.05-.05 1.95-.45 2.75-1.25S8.95 7.77 9 6.73h-.02C9.59 6.37 10 5.73 10 5zM2 1.8c.66 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2C1.35 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2zm0 12.41c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zm6-8c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"
          fillRule="evenodd"
        />
      </svg>
       master (joke)
    </a>
    <span
      className="footer-bar__info--lg"
    >
      <svg
        className="icon--footer"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 9H5l3-3 3 3H9v5H7V9zm5-4c0-.44-.91-3-4.5-3C5.08 2 3 3.92 3 6 1.02 6 0 7.52 0 9c0 1.53 1 3 3 3h3v-1.3H3c-1.62 0-1.7-1.42-1.7-1.7 0-.17.05-1.7 1.7-1.7h1.3V6c0-1.39 1.56-2.7 3.2-2.7 2.55 0 3.13 1.55 3.2 1.8v1.2H12c.81 0 2.7.22 2.7 2.2 0 2.09-2.25 2.2-2.7 2.2h-2V12h2c2.08 0 4-1.16 4-3.5C16 6.06 14.08 5 12 5z"
          fillRule="evenodd"
        />
      </svg>
    </span>
    <span
      className="footer-bar__info--lg"
      title="Active Linters"
    >
      <svg
        className="icon--footer"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 14 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6.3 5.69a.942.942 0 0 1-.28-.7c0-.28.09-.52.28-.7.19-.18.42-.28.7-.28.28 0 .52.09.7.28.18.19.28.42.28.7 0 .28-.09.52-.28.7a1 1 0 0 1-.7.3c-.28 0-.52-.11-.7-.3zM8 7.99c-.02-.25-.11-.48-.31-.69-.2-.19-.42-.3-.69-.31H6c-.27.02-.48.13-.69.31-.2.2-.3.44-.31.69h1v3c.02.27.11.5.31.69.2.2.42.31.69.31h1c.27 0 .48-.11.69-.31.2-.19.3-.42.31-.69H8V7.98v.01zM7 2.3c-3.14 0-5.7 2.54-5.7 5.68 0 3.14 2.56 5.7 5.7 5.7s5.7-2.55 5.7-5.7c0-3.15-2.56-5.69-5.7-5.69v.01zM7 .98c3.86 0 7 3.14 7 7s-3.14 7-7 7-7-3.12-7-7 3.14-7 7-7z"
          fillRule="evenodd"
        />
      </svg>
       ESLint, Standard
    </span>
    <span
      className="footer-bar__info--lg"
      title="0 Warning"
    >
      <svg
        baseProfile="tiny"
        className="icon--footer"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        version="1.2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 4c-4.419 0-8 3.582-8 8s3.581 8 8 8 8-3.582 8-8-3.581-8-8-8zm3.707 10.293c.391.391.391 1.023 0 1.414-.195.195-.451.293-.707.293s-.512-.098-.707-.293l-2.293-2.293-2.293 2.293c-.195.195-.451.293-.707.293s-.512-.098-.707-.293c-.391-.391-.391-1.023 0-1.414l2.293-2.293-2.293-2.293c-.391-.391-.391-1.023 0-1.414s1.023-.391 1.414 0l2.293 2.293 2.293-2.293c.391-.391 1.023-.391 1.414 0s.391 1.023 0 1.414l-2.293 2.293 2.293 2.293z"
        />
      </svg>
       0
       
      <svg
        className="icon--footer-small"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
        />
      </svg>
       0
    </span>
  </div>
  <div
    className="d-inline-block right footer-bar__right"
  >
    <span
      className="footer-bar__info--lg"
    >
      Spaces: 2
    </span>
    <span
      className="footer-bar__info--lg"
    >
      UTF-8
    </span>
    <span
      className="footer-bar__info--lg"
    >
      LF
    </span>
    <span
      className="footer-bar__info"
    >
      JavaScript
    </span>
    <span
      className="footer-bar__info"
    >
      Prettier:
      <svg
        className="icon--footer"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 12 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 5l-8 8-4-4 1.5-1.5L4 10l6.5-6.5L12 5z"
          fillRule="evenodd"
        />
      </svg>
    </span>
    <a
      className="reset-link footer-bar__info"
      href="https://github.com/vincentaudebert/vatweb/raw/master/public/assets/Vincent_Audebert_CV.pdf"
      rel="noopener noreferrer"
      target="_blank"
      title="Download my CV"
    >
      <svg
        className="icon--footer-small"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 496 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm80 168c17.7 0 32 14.3 32 32s-14.3 32-32 32-32-14.3-32-32 14.3-32 32-32zm-160 0c17.7 0 32 14.3 32 32s-14.3 32-32 32-32-14.3-32-32 14.3-32 32-32zm194.8 170.2C334.3 380.4 292.5 400 248 400s-86.3-19.6-114.8-53.8c-13.6-16.3 11-36.7 24.6-20.5 22.4 26.9 55.2 42.2 90.2 42.2s67.8-15.4 90.2-42.2c13.4-16.2 38.1 4.2 24.6 20.5z"
        />
      </svg>
    </a>
    <span
      aria-label="Bell"
      className="footer-bar__info--lg"
      role="img"
    >
      <svg
        className="icon--footer-small"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64zm215.39-149.71c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71z"
        />
      </svg>
    </span>
  </div>
</div>
`;
